remix-analyzer/src/solidity-analyzer/modules/abstractAstView.ts

```diff
@@ -46,23 +46,23 @@ export default class abstractAstView {
  * @contractsOut {list} return list for high level AST view
  * @return {ASTNode -> void} returns a function that can be used as visit function for static analysis modules, to build up a higher level AST view for further analysis.
  */
-  build_visit (relevantNodeFilter) {
-    var that = this
-    return function (node) {
-      if (isContractDefinition(node)) {
+  build_visit (relevantNodeFilter: ((node:any) => boolean)): VisitFunction {
+    const that: abstractAstView = this
+    return function (node: any) {
```

Comment:
By using an array function here you could get rid of the "that" `(node: any) => { ...this.setCurrentContract ... }`